homework-submission/photo-wall-app/src/components/post.js

```diff
@@ -0,0 +1,63 @@
+import React, { Component } from 'react'
+import PostsList from './posts-list'
+import AddForm from './add-from'
+import Service from './service'
+import Header from './header'
+
+class Post extends Component {
+  state = {
+    loading: false,
+    author: '',
+    posts: [],
+    form: 'list'
+  }
+
+  componentDidMount() {
+    this.setState({
+      loading: true,
+      author: this.props.state.user
+    })
+    Service.getPosts()
+      .then(posts => {
+        this.setState({ posts, loading: false })
+      })
+  }
+
+  handleAdd = (postInfo) => {
+    this.setState({
+      posts: [...this.state.posts, postInfo],
+      form: 'list'
+    })
+  }
+
+  switchToAddForm = () => {
+    this.setState({ form: 'add' })
+  }
+
+  switchToHome = () => {
+    this.setState({ form: 'list' })
+  }
+
+  render() {
+    console.log('postsState: ', this.state);
+    let lists = this.state.posts.map((post) => { return <PostsList key={post.id} posts={post} /> })
```

Comment:
can you find out why you get this warning, when you are add new photo:
Warning: Each child in an array or iterator should have a unique "key" prop.